christin/index.js

```diff
@@ -0,0 +1,273 @@
+}
+
+console.log("uniqueNumber==>", getUniqueNumber([0, 0, 0.77, 0, 0])); //
+
+//------------------------------------------------------------------------------------------
+
+/**5. Word Ranking. Create a function that takes a string of words and returns the highest scoring word. Each letter of a word scores points according to it's position in the alphabet: a = 1, b = 2, c = 3, etc.
+
+The returned string should only contain alphabetic characters (a-z).
+Preserve case in the returned string (see 4th example above).
+Examples:
+
+wordRank("The quick brown fox.") ➞ "brown"
+wordRank("Nancy is very pretty.") ➞ "pretty"
+wordRank("Check back tomorrow, man!") ➞ "tomorrow"
+wordRank("Today is Wednesday.") ➞ "Wednesday" */
+
+function WordRank(sentence) {
+  let abc = "abcdefghijklmnopqrstuvwxyz";
+
+  function getScoreOfWord(word) {
+    let sum = 0;
+    let strArr = word.split("");
+    strArr.forEach((chr) => (sum += abc.indexOf(chr) + 1));
+    return sum; //
+  }
+
+  let wordArray = sentence.split(" ");
+  let highestWord = wordArray.reduce((acc, word) => {
+    if (getScoreOfWord(word) > getScoreOfWord(acc)) {
+      acc = word;
+    }
+    return acc;
+  }, "");
+
+  return highestWord;
+}
+
+console.log(WordRank("The quick brown fox."));
+console.log(WordRank("Nancy is very pretty."));
+console.log(WordRank("Check back tomorrow, man!"));
+
+//------------------------------------------------------------------
+
+/**6. c4n y0u r34d th15? Create a function that takes a string as an argument and returns a coded (h4ck3r 5p34k) version of the string. NB: for your program to work properly, the function should replace all 'a's with 4, 'e's with 3, 'i's with 1, 'o's with 0, and 's's with 5.
+
+Examples:
+
+hackerSpeak("javascript is cool") ➞ "j4v45cr1pt 15 c00l"
+hackerSpeak("programming is fun") ➞ "pr0gr4mm1ng 15 fun"
+hackerSpeak("become a coder") ➞ "b3c0m3 4 c0d3r" */
+
+let getHacking = (string) => {
+  const vowels = {
+    a: 4,
+    e: 3,
+    i: 1,
+    o: 0,
+    s: 5,
+  };
+
+  let stringToArr = string.split("").map((el) => {
+    for (let key in vowels) {
+      if (el === key) {
+        el = vowels[key];
+      } else {
+        el = el;
+      }
+    }
+    return el;
+  });
+  return stringToArr.join("");
+};
+
+let hackerSpeak = getHacking;
+
+console.log("HackerSpeak=>", hackerSpeak("javascript is cool")); //j4v45cr1pt 15 c00l
+console.log("HackerSpeak2 ==>", hackerSpeak("programming is fun")); // pr0gr4mm1ng 15 fun
+
+//--------------------------------------------------------------------------------------------
+
+/* 1. Is it Symmetrical? Create a function that takes a number as an argument and returns true or false depending on whether the number is symmetrical or not. NB: A number is symmetrical when it is the same as its reverse.
+
+Examples:
+isSymmetrical(7227) ➞ true
+isSymmetrical(12567) ➞ false
+isSymmetrical(44444444) ➞ true
+isSymmetrical(9939) ➞ false
+isSymmetrical(1112111) ➞ true*/
+
+let isSymmetrical = (number) => {
```

Comment:
Bonus Task 1 ==> Output is always false! Even for Symmetrical numbers!
- In  justNmber variable get rid of .split("")
- In reverseNumber variable you can avoid to having duplicated code and write justNumber instead of number.toString() because you already stored number.toString() inside of justNumber
- The next step is to make a new variable and like for example const numReverse  and you have to store reverseNumber.join("") inside of this brand new variable.
- The last step is to to use numReverse instead of reverseNumber within if statement! 
- Now if  you try to check a Symmetrical number like 7227 you will get true!

Reply:
done.danke

Reply:
Sehr gerne:)